File: kata/persistence.rb
def persistence(n)
  num_string = n.to_s.split('').map(&:to_i)
  no_of_times = 0

  while num_string.size > 1
    final_value = num_string.reduce(:*)
    num_string = final_value.to_s.split('').map(&:to_i)
    no_of_times += 1
  end

  no_of_times
end

p persistence(39) == 3
p persistence(4) == 0
p persistence(25) == 2
p persistence(999) == 4

File: javascriptkata/persistence.js
function persistence(num) {
  var num = num.toString().split('').map(function(num) { return parseInt(num)});
  var no_of_times = 0;

  while (num.length > 1) {
    var final_value = 1
    num.forEach(function(each_value){
      final_value *= each_value;
    });
    num = final_value.toString().split('').map(function(num) { return parseInt(num)});
    no_of_times++;
  }

  return no_of_times;
}

console.log(persistence(39));
